src/registryDAO_storage.mligo

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2021 TQ Tezos
+// SPDX-License-Identifier: LicenseRef-MIT-TQ
+
```

Comment:
Not a big point, but since we have directories for variants (e.g. in this case `src/registryDAO`) wouldn't it be better for the storage definition to be moved in there?

I think that the same should be done for `template_storage.mligo` BTW

Reply:
I tried to do this but it didn't look good, because it would make the registry/treasury storage location to differ from what is said in the documentation, on how to define a variant. We can change the documentation and make it a little more complicated so that the user have to create a directory for the variant, and put the storage files there.

>I think that the same should be done for template_storage.mligo BTW

I didn't get this. Did you mean `template_storage.mligo` should be moved to `template` directory?

Reply:
I have new created a `variants` folder in `src` and have put `callbacks.mligo`, `storage.mligo` files in it. `callback.mligo` will be the `implementation.mligo` and `storage.mligo` will contact contract extra definitions. Similarly `templates` folder has been populated with template `callbacks.mligo` and `storage.mligo`. 

Does this look good?

Reply:
What I was suggesting was to move this file (`src/registryDAO_storage.mligo`) to `src/registryDAO/storage.mligo` given that we already had an `src/registryDAO` directory.

> it would make the registry/treasury storage location to differ from what is said in the documentation, on how to define a variant.

Yeah, that's why I suggested to do the same for `template_storage.mligo`.

AFAIU each variant needs 2 files now, I think we should put these in separate directories so that:
- all types and code related to a variant can be easily found and are organized
- anyone that wants to create their own custom DAO can copy the `template` directory and fill in the two files as they wish

what do you think? is that possible?

I see that you have already done some work in that direction in a sense, except for a few things that are out of place IMO:
1. there is a `src/registryDAO/types.mligo` which is a bit odd given that this is the only file in `src/registryDAO` and everything else about registry DAO seems to be under `src/variants/registry`. Can't the two directory be merged? Perhaps the content of the `types.mligo` can also be moved to `callbacks` or `storage`
2. same for `treasuryDAO`
3. I see that `template` now is out of `src`. Given that it's LIGO code I'd move it in `src/template` or maybe even `src/variants/template`. 

I think that changing these would improve the codebase structure, which LGTM for everything else.

Reply:
I have 

1. Moved `registryDAO/types.mligo` to `variants/registry/types.mligo`
2. And same for `treasury`.
3. Moved `template` to `src/variants`.

Will update the docs if this looks good.

Reply:
:+1: looks much better to me now, let's go with this